### PR TITLE
Added warning about "esm" support in docs

### PR DESCRIPTION
**Pull request by ozonep**
Discussed here: #653

### CODE REVIEW

File: docs/src/content/docs/using-with/index.md [hunk 1]
```diff
@@ -12,8 +12,14 @@ ES-Modules are supported by default, using the [esm](http://npm.im/esm) module.
...
-To turn _off_ ES-Module support (as may be required in some edge cases), run
-your tests with `--no-esm`.
+Although, if you are using ES-modules with `.js` extension & `type: module` pattern
+on Node.js v12.13+, you will face warnings and/or crashes while running tests. 
+In this case it is advised to turn "esm" support off & use native Node.js 
+module loader.
+
+To turn _off_ ES-Module support, run your tests with `--no-esm`.
+
+In future releases (v15+) "esm" module will be turned off by default.
```
Comment (coreyfarrell) [inline docs/src/content/docs/using-with/index.md:22]:
```suggestion
In future releases (v15+) "esm" module will not be provided by tap.
```
